File: algorithms/cpp/countSubstringsThatDifferbyOneCharacter/countSubstringsThatDifferbyOneCharacter.cpp
// Source: https://leetcode.com/problems/count-substrings-that-differ-by-one-character/
// Author: Miao Zhang
// Date:   2021-05-24

class Solution {
public:
    int countSubstrings(string s, string t) {
        int m = s.length();
        int n = t.length();
        int res = 0, diff = 0;
        for (int i = 0; i < m; ++i)
            for (int j = 0; j < n; ++j, diff = 0)
                for (int p = 0; i + p < m && j + p < n && diff <= 1; ++p)
                    if ((diff += (s[i + p] != t[j + p])) == 1) ++res;
        return res;
    }
};

File: algorithms/python/countSubstringsThatDifferbyOneCharacter/countSubstringsThatDifferbyOneCharacter.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/count-substrings-that-differ-by-one-character/
# Author: Miao Zhang
# Date:   2021-05-24

class Solution:
    def countSubstrings(self, s: str, t: str) -> int:
        m = len(s)
        n = len(t)
        res = 0
        for i in range(m):
            for j in range(n):
                diff = 0
                p = 0
                while i + p < m and j + p < n and diff <= 1:
                    diff += (s[i + p] != t[j + p])
                    if diff == 1:
                        res += 1
                    p += 1
        return res
